### DISCUSSION TITLE
How to set seasonal time program?

### DISCUSSION THREAD

**Discussion by citoki** [signalkraft/mypyllant-component#44] [Q&A] [answered]
I saw a [configuration sample](https://signalkraft.com/mypyllant-component/2-services/#setting-a-time-program) in the documentation for the [services](https://signalkraft.com/mypyllant-component/2-services/). In HomeAssitant it self I can only see the mentioned standard `myVaillant*` services. 

How can I add the custom service for setting the **time program**? Would it be possible to setup multiple time programs for different seasons or weather forecasts?

**Comment by signalkraft** [comment, marked as answer]
It's currently only available in the latest pre-release version. You could certainly switch between programs with HA automations, based on all kinds of data.

**Comment by Depechie** [comment]
Hey @signalkraft any tips/tricks on how to do this?
Sorry for the total noob questions.

Let's give an example, assume I want to have a different day schedule when working from home versus working at the office.
So I need to craft the JSON payloads per day of the week for remote and a fixed day schedule JSON payload for working from home.
But what would be the best thing to do... call the time service first with the [] payload to clear the day and then set the new payload?
I guess that has to be done best, prior to the start of the actual day... ( but that is straightforward with data and automations of course ).

Those JSON payloads are just files on disk?
